src/solve/SimulationParameters.h

```diff
@@ -74,6 +74,47 @@ struct SimulationParameters {
                                        ///< running coupled
 };
 
+/// @brief Wrapper class for nlohmann:json with error checking
+class JsonWrapper : public nlohmann::json {
+ public:
+  /**
+   * @brief Wrap around JSON configuration with detailed error message in case
+   * key is not found in configuration
+   *
+   * @param json JSON configuration
+   * @param name Name of the JSON configuration to be extracted
+   * @param id Index of JSON sub-list to be extracted
+   */
+  JsonWrapper(const nlohmann::json& json, const std::string& name,
+              const int& id)
+      : nlohmann::json(json[name][id]), component(name), block_id(id) {}
+
+  /**
+   * @brief Wrap error check around key retrieval (throws detailed error if key
+   * doesn't exist)
+   *
+   * @param key Key to retrieve from JSON object
+   * @return JSON entry of key
+   */
+  const nlohmann::json& operator[](const char* key) const {
+    if (!this->contains(key)) {
+      throw std::runtime_error(
+          "Key " + std::string(key) + " not found in element number " +
+          std::to_string(block_id) + " of component " + component);
```

Comment:
Is there a way to throw an error with the name of the element instead of the ID? For example, if you have a JSON file with 100 boundary conditions and there is a type in the 73rd one, it might be hard to find the error with only the index. Not a big deal if that is too hard to implement.

Reply:
It can definitely be done. The caveat is that the name could be corrupted (missing or wrong key), so we'd have to check for that before initializing `JsonWrapper` or in its constructor. 

I just noticed that we have different name keys for different categories (`vessel_name, bc_name, junction_name, name`), and `ClosedLoopHeartAndPulmonary` doesn't have a name - Same thing for `type`.

Both make reading out the name in the error message a bit tedious. Up to you if we want to include it!

Reply:
Yeah I guess the easiest way would be to have a `if (!config[component].contains(<name-key>)) {throw error}` before each `JsonWrapper` is initialized (except `ClosedLoopHeartAndPulmonary`). I'm fine with leaving it as is or adding this condition.